File: trading-analysis-platform/frontend/src/features/marketMovers/marketMoversTypes.ts
// Tipos del modulo de market movers (forma de la API /api/market-movers).

export interface MarketMoverDto {
  symbol: string;
  name: string | null;
  price: number | null;
  change: number | null;
  changePercent: number | null;
  volume: number | null;
  marketCap: number | null;
  ranking: number | null;
  source: string;
}

export interface MoversListDto {
  lastUpdated: string | null;
  items: MarketMoverDto[];
}

export interface AllMoversResponse {
  trending: MoversListDto;
  topGainers: MoversListDto;
  topLosers: MoversListDto;
  mostActive: MoversListDto;
  warnings: string[];
}

export const MOVERS_TABS = [
  { key: "trending", label: "Tendencia" },
  { key: "topGainers", label: "Mayores subidas" },
  { key: "topLosers", label: "Mayores caídas" },
  { key: "mostActive", label: "Más activas" },
] as const;

export type MoversTabKey = (typeof MOVERS_TABS)[number]["key"];

File: trading-analysis-platform/backend/app/services/market_movers/market_movers_types.py
"""Tipos comunes de market movers."""
from __future__ import annotations

from dataclasses import dataclass

LIST_TRENDING = "TRENDING"
LIST_TOP_GAINERS = "TOP_GAINERS"
LIST_TOP_LOSERS = "TOP_LOSERS"
LIST_MOST_ACTIVE = "MOST_ACTIVE"

ALL_LIST_TYPES = (LIST_TRENDING, LIST_TOP_GAINERS, LIST_TOP_LOSERS, LIST_MOST_ACTIVE)


@dataclass
class MarketMoverItem:
    symbol: str
    source: str
    name: str | None = None
    price: float | None = None
    change: float | None = None
    changePercent: float | None = None
    volume: float | None = None
    marketCap: float | None = None
    raw: dict | None = None
